### PR TITLE
Added an (optional) region param

### PR DESCRIPTION
**Pull request by SStorm**
## Summary of the changes / Why this is an improvement

This allows us to set a default region (which can also be __any__) and not have to ever specify one when running any commands.

## Checklist

 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed

### CODE REVIEW

**Comment by chaudum** [comment]
I like the change a lot! It will make the distinction between endpoint and region clearer.

However, I would only put not deprecated regions into the default profile: 

```yaml
default-format: table
current-profile: aks1.eastus2.azure
profiles:
  aks1.eastus2.azure:
    auth-token: NULL
    endpoint: https://console.cratedb.cloud
    region: aks1.eastus2.azure
  aks1.westeurope.azure:
    auth-token: NULL
    endpoint: https://console.cratedb.cloud
    region: aks1.westeurope.azure
  eks1.eu-west-1.aws:
    auth-token: NULL
    endpoint: https://console.cratedb.cloud
    region: eks1.eu-west-1.aws
```

Additionally, we need to update the documentation a bit, specifically [docs/configuration.rst](https://github.com/crate/croud/blob/master/docs/configuration.rst) and [docs/commands/regions.rst](https://github.com/crate/croud/blob/master/docs/commands/regions.rst). Maybe you want to check with @matthijskrul how we could pimp the information on regions.

**Comment by SStorm** [comment]
Have updated the docs and the profiles list to only show the non-deprecated ones. @matthijskrul docs feedback will be highly appreciated!

File: croud/api.py [hunk 1]
```diff
@@ -93,7 +93,7 @@ def from_args(args: Namespace) -> "Client":
             CONFIG.endpoint,
             token=CONFIG.token,
             on_token=CONFIG.set_current_auth_token,
-            region=args.region,
+            region=args.region if args.region is not None else CONFIG.region,
```
Comment (chaudum) [inline croud/api.py:96]: If you don't want to explicitly check for `None`, you could simpler write:

```suggestion
            region=args.region or CONFIG.region,
```

**Review by chaudum — APPROVED** [review]
Great stuff!

File: docs/configuration.rst [hunk 2]
```diff
@@ -59,12 +59,14 @@ The keys have the following meaning:
...
     (see :ref:`available-profiles`).
-    Each profile consists of ``auth-token``, ``endpoint``, and ``format``.
+    Each profile consists of ``auth-token``, ``endpoint``, ``region``, and ``format``.
 
     ``auth-token`` is populated with the API token upon login.
 
     ``endpoint`` is the full URL of the API endpoint that is used for requests.
 
+    ``region`` is the specific region where CrateDB Cloud resources are accessed in.
+
```
Comment (matthijskrul) [inline docs/configuration.rst:69]: I'd rephrase this: something like "``region`` is the specific region within which CrateDB Cloud resources are accessed".

File: docs/configuration.rst [hunk 3]
```diff
@@ -84,13 +86,15 @@ details.
...
 ==================
 
-==================== ====================================== ===========
-Profile              Endpoint                               Format
-==================== ====================================== ===========
-bregenz.a1           https://bregenz.a1.cratedb.cloud       table
-eastus2.azure        https://eastus2.azure.cratedb.cloud    table
-westeurope.azure     https://westeurope.azure.cratedb.cloud table
-==================== ====================================== ===========
+The API Endpoint for all profiles is ``https://console.cratedb.cloud``
```
Comment (matthijskrul) [inline docs/configuration.rst:89]: Should have a period at the end. Also I'd do lowercase "endpoint"

File: docs/commands/regions.rst [hunk 4]
```diff
@@ -15,19 +15,21 @@ Print the available regions to the user:
 
 .. note::
 
-   Listed region names can used in the ``--region`` argument to list/deploy resources in that region.
+   The region is specified for each profile in the :doc:`../configuration` file.
+   The region for specific actions can be overridden using the ``--region`` argument to list/deploy resources in that region.
```
Comment (matthijskrul) [inline docs/commands/regions.rst:19]: Preferably "list or deploy"
